metaflow/plugins/aws/secrets_manager/aws_secrets_manager_secrets_provider.py

```diff
@@ -0,0 +1,167 @@
+import base64
+import json
+from json import JSONDecodeError
+
+
+from metaflow.exception import MetaflowException
+from metaflow.metaflow_config import AWS_SECRETS_MANAGER_DEFAULT_REGION
+from metaflow.plugins.secrets import SecretsProvider
+import re
+
+
+class MetaflowAWSSecretsManagerBadResponse(MetaflowException):
+    """Raised when the response from AWS Secrets Manager is not valid in some way"""
+
+    pass
+
+
+class MetaflowAWSSecretsManagerDuplicateKey(MetaflowException):
+    """Raised when the response from AWS Secrets Manager contains duplicate keys"""
+
+    pass
+
+
+class MetaflowAWSSecretsManagerJSONParseError(MetaflowException):
```

Comment:
I am surprised you don't need "pass" here and below. May be safer to add though.